runtime/compiler/control/JITaaSCompilationThread.cpp

```diff
@@ -2603,19 +2591,28 @@ remoteCompile(
       resolvedMirrorMethodsPersistIPInfo = std::get<7>(recv);
       if (statusCode >= compilationMaxError)
          throw JITaaS::StreamTypeMismatch("Did not receive a valid TR_CompilationErrorCode as the final message on the stream.");
+      if (statusCode == compilationStreamVersionIncompatible)
+         throw JITaaS::StreamVersionIncompatible();
+      client->setVersionCheckStatus();
```

Comment:
Is this the only place where we need to set the status of the version check? At this point the entire compilation is finished, so the next compilation on the same stream will be fine, but how about all the other messages sent for this compilation? Will they have the version field set to 0 to avoid message bloat?

Reply:
`buildCompileRequest` sends version, and only the `buildCompileRequest` operation sends version. Regular `write` operation won't send version info.
A summary of how it operates:
`readCompileRequest` receives version and check, if fails, throws `StreamVersionIncompatible`, then server terminates the compilation and sends back a `compilationCode` message with `statusCode` being set to `compilationStreamVersionIncompatible`, client does not send any other messages to the server for this compilation if the version check failed. If pass, all remaining write operations do not send version.

Reply:
In `buildCompileRequest`:
we clear the version after we send the message, so that proceeding `write` operations won't have version set.
```
      if (getVersionCheckStatus() == NOT_DONE)
         {
         _cMsg.set_version(getJITaaSVersion());
         write(J9ServerMessageType::compilationRequest, args...);
         _cMsg.clear_version();
         }
```